typing/ctype.ml

```diff
@@ -3379,7 +3379,10 @@ let add_method env label priv virt ty sign =
           | Mprivate k ->
             match priv with
             | Public ->
-                link_kind ~inside:k field_public;
+                begin match field_kind_repr k with
+                | Fpublic | Fabsent -> ()
```

Comment:
I think this should still probably assert false in the `Fabsent` case. We should be unifying the kind variable with "present", which should be a no-op if it's already present (`Fpublic`), mutate it if it is currently a variable (`Fprivate`), and fail if it is already been unified with "absent" (`Fabsent`).

(I'm surprised this isn't already the behaviour of `link_kind`, hence the bug)

Reply:
Good point, failing on `Fabsent` seems better.